syntax = "proto3";

package largemessage;

// Service definition for testing large message handling
service LargeMessageService {
  // RPC method that handles large messages
  rpc ProcessLargeMessage(LargeMessage) returns (LargeMessageResponse) {}
  
  // RPC method for health check
  rpc HealthCheck(HealthRequest) returns (HealthResponse) {}
}

// Message containing large data
message LargeMessage {
  bytes data = 1;
  string message_id = 2;
  int64 timestamp = 3;
}

// Response for large message processing
message LargeMessageResponse {
  string status = 1;
  string message_id = 2;
  int64 processing_time_ms = 3;
  int64 data_size = 4;
}

// Health check request
message HealthRequest {
  string service_name = 1;
}

// Health check response
message HealthResponse {
  string status = 1;
  string timestamp = 2;
}
